src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/DefaultLSPDocumentSynchronizer.cs

```diff
@@ -44,9 +46,14 @@ public DefaultLSPDocumentSynchronizer(FileUriProvider fileUriProvider, LSPDocume
             _documentManager = documentManager;
         }
 
-        internal record SynchronizedResult<TVirtualDocumentSnapshot>(bool Synchronized, TVirtualDocumentSnapshot VirtualSnapshot)
+        internal record SynchronizedResult<TVirtualDocumentSnapshot>(bool Synchronized, TVirtualDocumentSnapshot? VirtualSnapshot)
             where TVirtualDocumentSnapshot : VirtualDocumentSnapshot
         {
+            public bool TryGetVirtualSnapshot([NotNullWhen(true)] out TVirtualDocumentSnapshot? virtualDocumentSnapshot)
```

Comment:
This is the weirdest `TryGet` pattern I've ever seen. You know you've just reinvented `Nullable<>`, right?

I mean, I don't mind it, its just a bit weird. Works though 🤷‍♂️